### DISCUSSION TITLE
Aborts when bind().

### DISCUSSION THREAD

**Discussion by Tosaaa** [ANLAB-KAIST/KENSv3#47] [Project Q&A 1. Application Layer (Socket)] [answered]
In serverMain method, I tried to bind my socket to given IP and port.
But when bind() is executed, the program aborts with the following error message.

echo: /home/tosa/KENSv3-master/src/Networking/E_Host.cpp:107: virtual E::Module::Message E::Host::messageReceived(E::ModuleID, E::Module::MessageBase&): Assertion `nsIter != appIter->second.fdToDomain.end()' failed.

I have checked every discussion here, and I tried to check if there is any memory bugs.
(https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Debugging-Memory-Bugs-(Only-for-macOS-and-Linux))
But I could not get any detailed memory bug message.
Will there be any solution to this problem?

current environment: vscode + wsl

**Comment by Pusnow** [comment]
This happens when you bind wrong socket. Check followings.

* Use `socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)`.
* Avoid defining/using custom functions.

**Reply by Tosaaa** [reply]
I opened socket with the given parameters and I didn't use any custom functions.
The return value of socket() is 3 and it seems valid file descriptor.

**Reply by Pusnow** [reply]
Please, send sources to TA. cs341.ta.pj@kaist.ac.kr

**Comment by Pusnow** [comment, marked as answer]
The problem was `EchoAssignment::clientMain`.

```c++
int serverfd;
  if (serverfd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP) < 0) {
    return -1;
  }
```

should be

```c++
int serverfd;
  if ((serverfd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) < 0) {
    return -1;
  }
```
